src/Microsoft.ServiceFabric.Actors/Runtime/KvsActorStateProviderBase.cs

```diff
@@ -972,6 +992,159 @@ private static object Deserialize(DataContractSerializer serializer, byte[] data
             }
         }
 
+        private async Task StartStateProviderInitializationAsync(CancellationToken cancellationToken)
+        {
+            Exception unexpectedException = null;
+
+            try
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                await this.actorStateProviderHelper.ExecuteWithRetriesAsync(
+                    async () =>
+                    {
+                        await this.InitializeAndStartLogicalTimeManagerAsync(cancellationToken);
+                    },
+                    "StartStateProviderInitializationAsync",
+                    cancellationToken);
+            }
+            catch (OperationCanceledException opEx)

```

Comment:
Same comment about handling TaskCanceledException. Please check VolatileActorStateProvider comments.

Reply:
TaskCanceledException inherits from the OperationCanceledException, we dont need to catch that separately.